content/docs/learn/typed-errors/from-either-to-raise.md

```diff
@@ -0,0 +1,181 @@
+---
+title: From Either to Raise
+sidebar_position: 5
+description: Learning about Raise from other FP ecosystems.
+---
+
+# From `Either` to `Raise`
+
+Typed errors in other functional ecosystems usually revolve around
+special types like `Either`, which provide the ability to describe
+computations that are successful or end in error. This style is
+fully supported by Arrow, but the DSL based around `Raise` usually
+results in nicer code. This small guide describes common patterns
+in the `Either` style and how they translate into `Raise`.
+
+In the following discussion, we assume the following functions:
+
+```kotlin
+fun f(n: Int): Either<Error, String>
+fun g(s: String): Either<Error, Thing>
+fun h(s: String): Either<Boo, Thing>
+fun Thing.summarize(): String
+```
+
+## Sequential composition
+
+The basic way to compose computations that may error out is by
+sequentially executing each of them, and finishing early if
+an error appears. This is done using `flatMap` — if the next
+computation may also fail — or `map` — if you need to apply
+a pure computation to the result, if available.
+
+Take this possible combination of the aforementioned functions:
+
+```kotlin
+fun tururu(n: Int): Either<Error, String> =
+  f(n).flatMap { s ->
+    g(s).map { t ->
+      t.summarize()
+    }
+  }
+```
+
+The translation into the `Raise` DSL describes the same sequence
+of steps, but using a more sequential style:
+
+```kotlin
+fun tururu(n: Int): Either<Error, String> = either {
+  val s = f(n).bind()
+  val t = g(s).bind()
+  t.summarize()
+}
+```
+
+There are two important steps in this translation. At the beginning of
+the function, we have the `either` builder, which (1) indicates that
+the result of the function is `Either` of some error type and (2)
+creates a new scope in which you can use the `Raise` DSL.
+
+The second step is that every time that you have an `Either` value
+resulting from another computation (like `f` and `g` above), you need
+to call `bind`. This "injects" any potential error value in the context,
+and continues normal execution otherwise.
```

Comment:
So, um, I think that means the bind suffix translates to 

```
if (return value of f() was of type Error) { 
  return it now 
}
else {
  leave it as is and go to the next line.
}
```

Is that accurate?  If it is then... holy cow `bind()` is a misleading now.  I hope I'm wrong, for that reason.

Reply:
This is more or less accurate. It's essentially a `flatMap`, where the "next thing to do" is the rest of the computation.

Why do you find it misleading?

Reply:
Because the only other place I have seen "bind" as a standard operation is Haskell, where it is effectively an enhanced pipe operator.  That is, `foo.bind(callable)`.  

What's shown here is not flatMap.  It's safe-unwrap.  While they may turn into the same thing after rewriting down in the engine, from a user perspective they are very different, which makes this super confusing.

A more self-descriptive method name would have been `safeUnwrap()` or `valueOrReturn()`.  (I realize it's rather late to change the method name at this point, but this is the sort of thing that will trip people up for years to come.  At the very least the odd naming can be addressed head on in docs, even if the API can't be adjusted.)

Reply:
There's an underlying reason for the name. In Kotlin, using `suspend` functions, you can actually get hold of the continuation of the function. Then you actually get something very similar to `flatMap`.

Anyway, I don't think the name is really what matters here; in order to use the Raise DSL you need to learn what `bind` does, and in which situations you should use it.

Reply:
Right.  I think if you can explain in the docs why it's named that (eg, parallels the bind() method in this other API), that will help with learning what it does and in what situations to use it.